File: erlang/pr010.erl
% Problem 10
%
% 08 February 2002
%
%
% The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
%
% Find the sum of all the primes below two million.
%
% 142913828922

-module(pr010).
-export([solve/0]).

solve() ->
	Sieve = sieve:start(),
	Primes = sieve:primes_to(Sieve, 2000000),
	P2 = [ P || P <- Primes, P < 2000000 ],
	sieve:stop(Sieve),
	lists:sum(P2). 

File: kotlin/src/pr010.kt
// Problem 10
//
// 08 February 2002
//
// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
//
// Find the sum of all the primes below two million.
//
// 142913828922

package org.davidb.euler

class Pr010 : EulerSolution<Long>() {
    override val answer = 142913828922L
    override val num = 10

    override fun solve() = Sieve().iterator().asSequence()
        .takeWhile { it < 2000000 }
        .map { it.toLong() }
        .sum()
}
